libnetwork/drivers/macvlan/macvlan_network.go

```diff
@@ -59,8 +59,15 @@ func (d *driver) createNetwork(config *configuration) (bool, error) {
 	for _, nw := range networkList {
 		if config.Parent == nw.config.Parent {
 			if config.ID != nw.config.ID {
-				return false, fmt.Errorf("network %s is already using parent interface %s",
-					getDummyName(stringid.TruncateID(nw.config.ID)), config.Parent)
+				if config.MacvlanMode == modePassthru || nw.config.MacvlanMode == modePassthru {
+					nwName := getDummyName(stringid.TruncateID(nw.config.ID))
+					name := getDummyName(stringid.TruncateID(config.ID))
+					return false, fmt.Errorf(
+						"macvlan network %s is already using parent interface %s and either %s or %s has mode passthru",
+						nwName, config.Parent, nwName, name,
+					)
+				}
+				continue
```

Comment:
The function `getDummyName()` is used to generate the name for a dummy link when no parent interface is specified - it just translates `foo` into `dm-foo`. So, for `nwName` - as the actual name doesn't seem to be stored by the driver, its short-id is just `stringid.TruncateID(nw.config.ID)`.

It might be worth making the error message simpler and more specific by splitting out the two cases...

```suggestion
                                if config.MacvlanMode == modePassthru {
                                        return false, fmt.Errorf(
                                                "cannot use mode passthru, macvlan network %s is already using parent interface %s",
                                                stringid.TruncateID(nw.config.ID),
                                                config.Parent,
                                        )
                                } else if nw.config.MacvlanMode == modePassthru {
                                        return false, fmt.Errorf(
                                                "macvlan network %s is already using parent interface %s in mode passthru",
                                                stringid.TruncateID(nw.config.ID),
                                                config.Parent,
                                        )
                                }
                                continue
```